rholang/src/main/scala/coop/rchain/rholang/interpreter/storage/implicits.scala

```diff
@@ -0,0 +1,42 @@
+package coop.rchain.rholang.interpreter.storage
+
+import cats.implicits._
+import coop.rchain.models.{Channel, Par}
+import coop.rchain.models.Channel.ChannelInstance.Quote
+import coop.rchain.storage.{Match => StorageMatch}
+import coop.rchain.rholang.interpreter.SpatialMatcher._
+
+import scala.annotation.tailrec
+
+//noinspection ConvertExpressionToSAM
+object implicits {
+
+  implicit def orderingIntTuple[T]: Ordering[(Int, T)] =
+    new Ordering[(Int, T)] {
+
+      def compare(x: (Int, T), y: (Int, T)): Int =
+        (x, y) match {
+          case ((i1, _), (i2, _)) => i1.compare(i2)
+        }
+    }
+
+  private def toQuotes[T](fm: FreeMap)(implicit ord: Ordering[(Int, Par)]): Option[List[Quote]] = {
+    @tailrec
+    def loop(sortedParList: List[(Int, Par)], curr: Int, acc: List[Quote]): Option[List[Quote]] =
+      sortedParList match {
+        case Nil                         => Some(acc)
+        case (i, t) :: tail if i == curr => loop(tail, i + 1, Quote(t) :: acc)
```

Comment:
What is the purpose of tracking the `i`? Don't we in the end just get rid of that information and just return the List of Quotes?

Reply:
I'm curious about the why too. 

@guardbotmk3?

Reply:
So I can see that it is useful to return from the matcher if there was another Tuplespace implementation that didn't assume the order that the quotes were returned. But since the Tuplespace implementation you wrote just assumes to the order, I think we can just immediately drop it as in

def loop(sortedParList: List[(Int, Par)]) = sortedParList.map(Quote(_._2))

Reply:
Hmm actually we need to know how to then substitute.

Reply:
Are you sure we can just return Option[List[Quote]] for 

def get(patterns: List[Channel], data: List[Quote]): Option[List[Quote]]

. I think it might be handy to pass through the level info back to the interpreter.

Reply:
The point is that each free variable in the pattern needs an entry in the list.
Unfortunately freeCount isn't exactly what you want, since I rolled wildcards into it.
Can you @henrytill or @KentShikama write a small PR that splits out freecount from a wildcard boolean.

Reply:
patterns.map(freeCount).sum